File: src/main/java/com/methodpark/cmpe/code/srp/CsvReader.java
package com.methodpark.cmpe.code.srp;

import java.io.File;
import java.io.IOException;
import java.net.URI;
import java.nio.charset.Charset;
import java.nio.file.Files;
import java.util.ArrayList;
import java.util.List;

public class CsvReader
{
    private URI filename;

    private String[] header;

    private List<String[]> records = new ArrayList<String[]>();

    public CsvReader(URI filename)
    {
        this.filename = filename;

        parseFile();
    }

    public String[] getHeader()
    {
        return header;
    }

    public List<String[]> getRecords()
    {
        return records;
    }

    private void parseFile()
    {
        for (String line : readAllLinesOfFile())
        {
            if (header == null)
            {
                parseHeader(line);
            }
            else
            {
                String[] record = parseRecord(line);
                records.add(record);
            }
        }
    }

    private List<String> readAllLinesOfFile()
    {
        List<String> lines = new ArrayList<String>();
        File file = new File(filename);
        try
        {
            lines.addAll(Files.readAllLines(file.toPath(),
                    Charset.defaultCharset()));
        }
        catch (IOException e)
        {
            e.printStackTrace();
        }
        return lines;
    }

    private void parseHeader(String line)
    {
        header = parseRecord(line);
    }

    private String[] parseRecord(String line)
    {
        return line.split(",");
    }
}

File: src/main/csharp/Srp/Srp/CsvReader.cs
using System.Collections.Generic;
using System.IO;

namespace Srp
{
    public class CsvReader
    {
        private readonly string filename;

        public string[] Header { get; private set; }

        public List<string[]> Records
        {
            get { return records; }
        }

        private readonly List<string[]> records = new List<string[]>();

        public CsvReader(string filename)
        {
            this.filename = filename;

            ParseFile();
        }

        private void ParseFile()
        {
            foreach (var line in ReadAllLinesOfFile())
            {
                if (Header == null)
                {
                    ParseHeader(line);
                }
                else
                {
                    var record = ParseRecord(line);
                    records.Add(record);
                }
            }
        }

        private IEnumerable<string> ReadAllLinesOfFile()
        {
            return File.ReadAllLines(filename);
        }

        private void ParseHeader(string line)
        {
            Header = ParseRecord(line);
        }

        private static string[] ParseRecord(string line)
        {
            return line.Split(',');
        }
    }
}